manifests/certificate/x509.pp

```diff
@@ -123,8 +123,8 @@
   Optional[String]                   $state = undef,
   Optional[String]                   $locality = undef,
   Optional[String]                   $unit = undef,
-  Array                              $altnames = [],
-  Array                              $extkeyusage = [],
+  Optional[Array]                    $altnames = undef,
+  Optional[Array]                    $extkeyusage = undef,
   Optional[String]                   $email = undef,
```

Comment:
This is not solving the problem. It still allows to add an empty array to those parameters which will lead to the same result.